src/precice/tests/SerialTests.cpp

```diff
@@ -1661,6 +1661,80 @@ BOOST_AUTO_TEST_CASE(testSummationActionTwoSources)
   testSummationAction(configFile, context);
 }
 
+/**
+ * @brief
+ *
+ */
+void testConvergenceMeasures(const std::string configFile, TestContext const &context, std::vector<int> &expectedIterations)
+{
+  using Eigen::Vector2d;
+  using namespace precice::constants;
+
+  std::string meshName = context.isNamed("SolverOne") ? "MeshOne" : "MeshTwo";
+
+  SolverInterface cplInterface(context.name, configFile, 0, 1);
+  const int       meshID = cplInterface.getMeshID(meshName);
+
+  Vector2d vertex{0.0, 0.0};
+
+  std::vector<double> writeValues = {1.0, 1.01, 2.0, 2.5, 2.8, 2.81};
+
+  int vertexID = cplInterface.setMeshVertex(meshID, vertex.data());
+
+  cplInterface.initialize();
+  int numberOfAdvanceCalls = 0;
+  int numberOfIterations   = -1;
+  int timestep             = 0;
+
+  while (cplInterface.isCouplingOngoing()) {
+    if (cplInterface.isActionRequired(actionWriteIterationCheckpoint())) {
+      numberOfIterations = 0;
+      cplInterface.markActionFulfilled(actionWriteIterationCheckpoint());
+    }
+
+    if (context.isNamed("SolverTwo")) {
+      int dataID = cplInterface.getDataID("Data2", meshID);
+      cplInterface.writeScalarData(dataID, vertexID, writeValues[numberOfAdvanceCalls]);
+    }
+
+    cplInterface.advance(1.0);
+    ++numberOfAdvanceCalls;
+    ++numberOfIterations;
+
+    if (cplInterface.isActionRequired(actionReadIterationCheckpoint())) {
+      cplInterface.markActionFulfilled(actionReadIterationCheckpoint());
+    } else { //converged
+      BOOST_TEST(numberOfIterations == expectedIterations[timestep]);
```

Comment:
This is undefined behaviour and a SEGFAULT if there are more timesteps then provided.

Using `.at()` throws an exception:
```suggestion
      BOOST_TEST(numberOfIterations == expectedIterations.at(timestep));
```